hw/ip/otbn/rtl/otbn_controller.sv

```
@@ -1135,6 +1135,7 @@ module otbn_controller
 
   assign rf_d_indirect_err = insn_dec_bignum_i.rf_d_indirect    &
                              (|rf_base_rd_data_b_no_intg[31:5]) &
+                             ~rf_base_call_stack_sw_err_i       &
```

The second thing I wanted to clarify is if we are missing the check that `rf_base_rd_data_b_no_intg` indeed comes from the call stack before ignoring the error?

I think the condition would simply be `insn_dec_base_i.b == 5'd1`?

I have not added it from the beginning since `rf_a_indirect_err` and `rf_b_indirect_err` (just above) also ignore the upper bits of the register index when a call stack error occurs. But maybe the register index checks are missing there as well?

---

The suggested fix is the best approach I think. Two observations:

1. The indirect register error only makes sense if the the data from the register file is valid, so suppressing the error that's based on that data when we've had a call stack error makes sense
2. We don't want to check the indirect access is coming from the call stack register as we are doing two indirect accesses and a call stack error in either stops the instruction executing and architecturally it is stopped before we look at the base rf reads to do the bignum rf read (See the pseudo code https://docs.opentitan.org/hw/ip/otbn/doc/isa/#bnlid). So we don't want an `rf_d_indirect_err` if the other indirect access has seen a call stack error.

---

Got it, thanks for your explanation @GregAC!